src/Sylius/Component/Review/Calculator/AverageRatingCalculator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Review\Calculator;
+
+use Sylius\Component\Review\Model\ReviewableInterface;
+use Sylius\Component\Review\Model\ReviewInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class AverageRatingCalculator implements AverageRatingCalculatorInterface
+{
+    /**
+     * @var int
+     */
+    private $reviewsNumber = 0;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function calculate(ReviewableInterface $reviewable)
+    {
+        if (0 === count($reviews = $reviewable->getReviews())) {
+            return 0;
```

Comment:
I think this second ``if`` can be removed, because all calculating logic is unneeded if there are no reviews.